### ISSUE TITLE
Date Formatting Issues

### ISSUE THREAD

**Issue by Stitch10925** [joplin/plugin-templates#87]
The date is not formatted correctly. According to the documentation:

> NOTE: All dates are formatted based on your Joplin settings in General > Date Format

But I have my date format set to YYYY-MM-DD, but it still formats new notes as DD/MM/YYYY as described by my language settings which is set to: "English (United Kingdom) 100%"

Alternatively I tried to use the custom date formatting shown in the documentation, but I can't figure out how it works. An example might help. I tried:

`{{#YYYY-MM-DDDD}}`
`d{{/YYYY-MM-DDDD}}}}`
`{{#YYYY-MM-DDDD}}M`

But I just get errors.

**Comment by nishantwrp** [comment]
`{{#custom_datetime}}YYYY-MM-DD{{/custom_datetime}}` should work.

**Comment by Stitch10925** [comment]
> `{{#custom_datetime}}YYYY-MM-DD{{/custom_datetime}}` should work.

Ah, yes, thank you!

**Comment by unode** [comment]
@nishantwrp sorry to reopen an old thread but I am also encountering this on Linux.
All the documentation I found seems to indicate that the template dialog should respect the `Tools > Options > Date format` which I have set to `YYYY-MM-DD`, but it doesn't.

(Is this a UI bug?)

In the template I have:

```
---
start_date: date
end_date: date
```

<img alt="Image" src="https://github.com/user-attachments/assets/db956983-956c-4adb-9ea1-a0772334916f" />

I have tried to set `LC_TIME` and to change the date format and language of the interface but neither seem to have an effect.

I also tried to replace `start_date: date` with other variants, including `custom_datetime` but none were supported types.

Is there any other way to ensure the template shows a date format consistent with the settings?
